h2/src/test/org/h2/test/jdbc/TestResultSet.java

```diff
@@ -1949,13 +1949,7 @@ private void testEnum() throws SQLException {
         prep.setObject(2, "D", Types.VARCHAR);
         prep.executeUpdate();
         prep.setInt(1, 5);
-        prep.setObject(2, "E", Types.OTHER);
-        prep.executeUpdate();
-        prep.setInt(1, 6);
-        prep.setObject(2, 5, Types.OTHER);
-        prep.executeUpdate();
-        prep.setInt(1, 7);
-        prep.setObject(2, 6, Types.INTEGER);
+        prep.setObject(2, 4, Types.INTEGER);
```

Comment:
if 
    setObject(foo, enum or geometry, Types.OTHER)
that is an API change, and will quite likely break some client code

Reply:
I introduced such support in e41b44458b85a5b7d10df966904dd306f6ed2f17 and that change isn't related in any way with the original issue about `ENUM`.

On the other hand, the `Types.OTHER` should not be the same as `Types.JAVA_OBJECT`. `setObject()` with `Types.OTHER` should not be mapped to `ValueJavaObject` like it currently does. `OTHER` is just a placeholder for database-specific data types. It isn't needed since JDBC 4.2, because in 4.2 database can provide own data types.

Perhaps H2 should not create `ValueJavaObject` for `Types.OTHER` if more specific data type can be determined from the passed value.

Reply:
Let's continue this discussion in #2416.

Reply:
I restored support of `Types.OTHER` for `GEOMETRY` columns.